doc/wait-for-ready.md

```diff
@@ -1,14 +1,26 @@
 gRPC Wait for Ready Semantics
 =============================
 
-If an RPC is issued but the channel is in `TRANSIENT_FAILURE` or `SHUTDOWN`
-states, the RPC is unable to be transmited promptly. By default, gRPC
-implementations SHOULD fail such RPCs immediately. This is known as "fail fast,"
-but usage of the term is historical. RPCs SHOULD NOT fail as a result of the
-channel being in other states (`CONNECTING`, `READY`, or `IDLE`).
-
-gRPC implementations MAY provide a per-RPC option to not fail RPCs as a result
-of the channel being in `TRANSIENT_FAILURE` state. Instead, the implementation
-queues the RPCs until the channel is `READY`. This is known as "wait for ready."
-The RPCs SHOULD still fail before `READY` if there are unrelated reasons, such
-as the channel is `SHUTDOWN` or the RPC's deadline is reached.
+Errors can occur in many places. For the propose of this document we can bucket the errors into three separate situations:
+
+1. The RPC never leaves the gRPC client library, for example:
+    * The connection was broken and the channel is in a `TRANSIENT_FAILURE` state.
+2. The RPC reaches the gRPC server library, and is rejected. For example:
+    * The server is momentarily overloaded and sends a GOAWAY frame.
+3. The RPC reaches the server application, and fails.
+
+![Where RPCs Fail](images/WhereRPCsFail.png)
+
+In all of these cases, by default, gRPC implementations SHOULD immediately return the error to the client application. This is known as "fail fast" but the usage of the term is historical.
+
+gRPC implementations MAY provide a per-RPC option to not fail RPCs as a result of reason 1 and 2 (the cases in which the RPC never made it to the server application). This is known as "wait for ready" behavior. An RPC with this behavior enabled will stay pending on the client until:
+  * it sees an *irrecoverable* failure
+  * its deadline is reached
+
+Recalling the examples at top:
+  * an overloaded server may recover
```

Comment:
Let's not talk about "why" the server is doing something. Instead, this example should be reworked to talk about what the client might do when it receives a GOAWAY. (i.e., the client might be able to get a new working connection ).